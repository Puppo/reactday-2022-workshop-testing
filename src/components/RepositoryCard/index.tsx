interface Props {
  name: string;
  description: string;
}

export default function RepositoryCard({
  name, description
}: Props) {
  return <div className='card'>
    <div className="card-content">
      <div className="media">
        <div className="media-content">
          <p className="title is-4">{name}</p>
        </div>
      </div>
      <div className="content">
        {description}
      </div>
    </div>
  </div>
}
