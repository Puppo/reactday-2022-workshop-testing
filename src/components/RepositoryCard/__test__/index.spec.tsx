import { render, screen } from "@testing-library/react";
import RepositoryCard from "..";

describe('RepositoryCard', () => {

  it('should render the component', () => {
    const { container } = render(<RepositoryCard name="test" description="test" />);
    expect(container).toMatchSnapshot();
  });

  it('should render the repository name', () => {
    render(<RepositoryCard name="test-name" description="test-description" />);
    const el = screen.getByText('test-name')
    expect(el).toBeInTheDocument();
    expect(el.tagName).toBe('P');
  });

  it('should render the repository description', () => {
    render(<RepositoryCard name="test-name" description="test-description" />);
    const el = screen.getByText('test-description')
    expect(el).toBeInTheDocument();
    expect(el.tagName).toBe('DIV');
  });


})

export { };
