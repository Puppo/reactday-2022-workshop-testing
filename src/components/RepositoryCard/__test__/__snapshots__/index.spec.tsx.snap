// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCard should render the component 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-content"
    >
      <div
        class="media"
      >
        <div
          class="media-content"
        >
          <p
            class="title is-4"
          >
            test
          </p>
        </div>
      </div>
      <div
        class="content"
      >
        test
      </div>
    </div>
  </div>
</div>
`;
